test/parallel/test-http-parser-consume.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+const spawn = require('child_process').spawn;
+
+if (process.argv[2] === 'child') {
+  var server = http.createServer(function(q, s) {
+    s.end('hello');
+  });
+
+  server.listen(common.PORT, function(s) {
+    var rr = http.get('http://localhost:' + common.PORT, function(d) {
+      rr.parser.consume(0);
+      process.exit(0);
+    });
+  });
+}
+else {
+  const child = spawn(process.argv[0], [process.argv[1], 'child']);
+  child.on('stdout', function(d) {
```

Comment:
Sorry, don't know how I picked up 'stdout' as an event. Removed them and inherited from parent.

Debugging issues and isolating them between test issues and node issues have been effortful, so please allow me to retain the child output, if it is fine by you?